src/foam/nanos/http/LiveScriptBundler.java

```diff
@@ -0,0 +1,396 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.http;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
+import foam.core.ContextAware;
+import foam.core.X;
+import foam.nanos.logger.Logger;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.*;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.StandardWatchEventKinds.*;
+
+public class LiveScriptBundler implements WebAgent, ContextAware
```

Comment:
You could extend ContextAwareSupport so you don't have to implement ContextAware directly.